src/librustc/traits/error_reporting.rs

```diff
@@ -559,6 +559,28 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
                             trait_ref.to_predicate(),
                             post_message);
 
+                        let trait_display = trait_ref.to_string();
+                        if trait_display.starts_with("std::cmp::PartialEq") {
```

Comment:
This will not work with `no_std` ([as pointed out before to me](https://github.com/rust-lang/rust/pull/40565#discussion_r106342321)).